extensions/amp-ad-exit/0.1/amp-ad-exit.js

```diff
@@ -100,17 +103,68 @@ export class AmpAdExit extends AMP.BaseElement {
       'CLICK_Y': true,
     };
     if (target.vars) {
-      for (const customVar in target.vars) {
-        if (customVar[0] == '_') {
-          vars[customVar] = () =>
-              args[customVar] || target.vars[customVar].defaultValue;
-          whitelist[customVar] = true;
+      const all3pResponses = this.getAmpDoc().win[AMP_ANALYTICS_3P_RESPONSES] ||
+          {};
+
+      for (const customVarName in target.vars) {
+        if (customVarName[0] == '_') {
+          const customVar =
+              /** @type {./config.Variable} */ (target.vars[customVarName]);
+          if (customVar) {
+            /*
+              Example:
+              The amp-ad-exit target has a variable representing the
+               priority of something, which is defined as follows:
+               "vars": {
+                 "_pty": {
+                   "defaultValue": "unknown",
+                   "vendorAnalyticsSource": "vendorXYZ",
+                   "vendorAnalyticsResponseKey": "priority"
+                 },
+                 ...
+               }
+               The cross-domain iframe of vendorXYZ has sent the
+               following response for the creative:
+                 { priority: medium, category: W }
+               This is just example data. The keys/values in that object can
+               be any strings.
+               The code below will create substitutionFunctions['_pty'],
+               which in this example will return "medium".
+             */
+            substitutionFunctions[customVarName] = () => {
```

Comment:
would this overwrite `CLICK_X` `CLICK_Y`, or we allow them to overwrite these two functions.

Reply:
It would not be allowed to overwrite CLICK_X or CLICK_Y, since we earlier verified that customVarName begins with '_'.

Reply:
Ah I see. It took me a while to find the underscore requirement in the document even after you told me this : (
Do we have checks for this? in `config.js` when we assert Configs.

Reply:
We do check this in config.js, but the error message and regex name could be improved:
https://github.com/ampproject/amphtml/blob/master/extensions/amp-ad-exit/0.1/config.js#L136